### PR TITLE
Test

### PR DESCRIPTION
**Pull request by OlesiaSemen**

### CODE REVIEW

File: src/checkPassword.test.js [hunk 1]
```diff
@@ -12,8 +12,51 @@ describe(`Function 'checkPassword':`, () => {
...
+  });
+
+  it('should return true for a valid password', () => {
+    const result = checkPassword('Password1!');
+
+    expect(result).to.be.true;
+  });
+
+  it('should return false for a password without uppercase letters', () => {
+    const result = checkPassword('qwerty');
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: missing spec char and digit

File: src/checkPassword.test.js [hunk 2]
```diff
@@ -12,8 +12,51 @@ describe(`Function 'checkPassword':`, () => {
...
+
+    expect(result).to.be.false;
+  });
+
+  it('should return false for a password without special characters', () => {
+    const result = checkPassword('StrongPassword');
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: missing digit

File: src/checkPassword.test.js [hunk 3]
```diff
@@ -12,8 +12,51 @@ describe(`Function 'checkPassword':`, () => {
...
+  });
+
+  it('should return false for a password without digits', () => {
+    const result = checkPassword('Str@ng');
+
+    expect(result).to.be.false;
+  });
+
+  it('should return false for a short password', () => {
+    const result = checkPassword('Pwd1!');
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: boundary is 7

File: src/checkPassword.test.js [hunk 4]
```diff
@@ -12,8 +12,51 @@ describe(`Function 'checkPassword':`, () => {
...
+
+    expect(result).to.be.false;
+  });
+
+  it('should return false for a long password', () => {
+    const result = checkPassword('VeryLongPassword123!@#$%^');
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: boundary is 17

File: src/checkPassword.test.js [hunk 5]
```diff
@@ -12,8 +12,51 @@ describe(`Function 'checkPassword':`, () => {
...
+
+    expect(result).to.be.false;
+  });
+
+  it('should return false for a password with invalid characters', () => {
+    const result = checkPassword('Invalid*Password1');
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: why this input is invalid?

File: src/checkPassword.test.js [hunk 6]
```diff
@@ -12,8 +12,51 @@ describe(`Function 'checkPassword':`, () => {
   });
 
   it(`should return 'true' for the valid password with 8 characters`, () => {
+    const result = checkPassword('Pass123!@');
 
+    expect(result).to.be.true;
+  });
+
+  it('should return true for a valid password', () => {
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: what about the boundary values technique? add tests

**Review by 4ksandrey — CHANGES_REQUESTED** [review]
Check the comments

**Review by 4ksandrey — CHANGES_REQUESTED** [review]
No changes, please make sure that you commit your changes

File: src/checkPassword.test.js [hunk 7]
```diff
@@ -12,8 +12,52 @@ describe(`Function 'checkPassword':`, () => {
...
+  });
+
+  it('should return false for a password with 7 characters', () => {
+    const result = checkPassword('Pwd1hj!');
+
+    expect(result).to.be.false;
+  });
+
+  it('should return false for a password with 17 characters', () => {
+    const result = checkPassword('VeryLongPassword1');
```
Comment (4ksandrey) [inline src/checkPassword.test.js:52]: missing special char

File: src/checkPassword.test.js [hunk 8]
```diff
@@ -12,8 +12,52 @@ describe(`Function 'checkPassword':`, () => {
...
 
   it(`should return 'true' for the valid password with 8 characters`, () => {
+    const result = checkPassword('Pass123!@');
 
+    expect(result).to.be.true;
+  });
+
+  
+  it('should return false for a password contains at least 1 digit, 1 special character, 1 uppercase letter.', () => {
+    const result = checkPassword('qwerty');
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: not relevant test, your negative inputs should have only one deviation from the requirements

**Review by 4ksandrey — CHANGES_REQUESTED** [review]
Check the comments

File: src/checkPassword.test.js [hunk 9]
```diff
@@ -12,8 +12,62 @@ describe(`Function 'checkPassword':`, () => {
...
+  });
+
+  it('should return true for a valid password', () => {
+    const result = checkPassword('Password1!');
+
+    expect(result).to.be.true;
+  });
+
+  it('should return false for a password without special characters', () => {
+    const result = checkPassword('StrongPassword');
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: this input should include digit

File: src/checkPassword.test.js [hunk 10]
```diff
@@ -12,8 +12,62 @@ describe(`Function 'checkPassword':`, () => {
...
+    const result = checkPassword('Pwd1hj!');
+
+    expect(result).to.be.false;
+  });
+
+  it('should return false for a password with 17 characters', () => {
+    const result = checkPassword('VeryLongPassword1');
+
+    expect(result).to.be.false;
+  });
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: duplicates

File: src/checkPassword.test.js (hunk 5 above)
Comment (4ksandrey) [reply src/checkPassword.test.js]: not fixed

**Review by 4ksandrey — CHANGES_REQUESTED** [review]
Check the comments

File: src/checkPassword.test.js (hunk 9 above)
Comment (Ivan-Shvorak) [reply src/checkPassword.test.js]: not fixed, please make sure that all negative tests have only 1 deviation from the requirements

File: src/checkPassword.test.js [hunk 11]
```diff
@@ -12,8 +12,50 @@ describe(`Function 'checkPassword':`, () => {
...
+    const result = checkPassword('Pass2!@');
+
+    expect(result).to.be.false;
+  });
+
+  it(`should return false for the password with 17 characters`, () => {
+    const result = checkPassword('Password123ijhda!');
+
+    expect(result).to.be.false;
+  });
```
Comment (Ivan-Shvorak) [inline src/checkPassword.test.js:30]: add empty line between tests

File: src/checkPassword.test.js [hunk 12]
```diff
@@ -12,8 +12,50 @@ describe(`Function 'checkPassword':`, () => {
...
+  it(`should return true for the password with 16 characters including at least 1 digit, 1 special character, 1 uppercase letter. `, () => {
+    const result = checkPassword('Passwordwertdc!1');
+
+    expect(result).to.be.false;
+  });
+
+  it('should return true for a valid password', () => {
```
Comment (Ivan-Shvorak) [inline src/checkPassword.test.js]: it is the same as the 1st test

**Review by Ivan-Shvorak — CHANGES_REQUESTED** [review]
Please check the comments

**Review by 4ksandrey — APPROVED** [review]
Accepted, well done
